pkg/cloudprovider/provider/azure/types/types.go

```diff
@@ -38,11 +36,18 @@ type RawConfig struct {
 	AvailabilitySet   providerconfigtypes.ConfigVarString `json:"availabilitySet"`
 	SecurityGroupName providerconfigtypes.ConfigVarString `json:"securityGroupName"`
 	Zones             []string                            `json:"zones"`
-	ImagePlan         *compute.Plan                       `json:"imagePlan"`
+	ImagePlan         *imagePlan                          `json:"imagePlan"`
 
 	ImageID        providerconfigtypes.ConfigVarString `json:"imageID"`
 	OSDiskSize     int32                               `json:"osDiskSize"`
 	DataDiskSize   int32                               `json:"dataDiskSize"`
 	AssignPublicIP providerconfigtypes.ConfigVarBool   `json:"assignPublicIP"`
 	Tags           map[string]string                   `json:"tags,omitempty"`
 }
+
+// imagePlan contains azure OS Plan fields for the marketplace images
+type imagePlan struct {
```

Comment:
I suppose this structure should be public

Reply:
why ?

Reply:
how otherwise would you programmatically setup ImagePlan field?